kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/model/S3WriteMode.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 Lenses.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lenses.streamreactor.connect.aws.s3.model
+
+import enumeratum.{Enum, EnumEntry}
+
+import scala.collection.immutable
+
+sealed trait S3WriteMode extends EnumEntry
+
+object S3WriteMode extends Enum[S3WriteMode] {
+
+  override val values: immutable.IndexedSeq[S3WriteMode] = findValues
+
+  case object Streamed extends S3WriteMode
```

Comment:
These two should be able to create the location and have

```
sealed trait Location {
  def path:String
}

case class Local (path:java.nio.Path) extends Location
case class Remote(path: String) extends Location 
```

Reply:
This wouldn't work as overriding `def path:String` with `path:java.nio.Path`, however I've followed the majority of the suggestion.